api/types.proto

```diff
@@ -447,13 +447,25 @@ message AcceptancePolicy {
 }
 
 
+message ExternalCA {
+	enum CAType {
+		CFSSL = 0 [(gogoproto.enumvalue_customname) = "CATypeCFSSL"];
+	}
+
+	// CAType is the type of this external CA.
+	CAType type = 1;
+
+	// URL is the URL where the external CA can be reached.
+	string url = 2 [(gogoproto.customname) = "URL"];
```

Comment:
@aaronlehmann I think the best option is probably to support arbitrary key-value string pairs and let the code for any given external CA type decide how those translate into headers, query parameters, POST data, or anything else necessary for that particular CA.